### ISSUE TITLE
Suggest snippets only for "new" lines

### ISSUE THREAD

**Issue by sQVe** [hrsh7th/nvim-cmp#276]
👋🏼,

One annoyance I'm having is that I sometime get snippet suggestions when they really shouldn't be suggested. Is it possible to somehow configure snippets to only be valid as suggestions in specific use cases?

An example is that `arr.<cursor>` isn't a valid location but the follow is:
```typescript
function() {
  <cursor>
}
```

I'm using `vsnip`, if that makes any difference.

**Comment by sQVe** [comment]
🤔 

I tried this but it doesn't seem to work:

```lua
{name = 'vsnip', priority = 40, keyword_pattern = [[^\s*]]}
```

**Comment by sQVe** [comment]
Hmm, no - that doesn't work either. Really confused on how to get this to work. Using `substitute` works fine with the vim regex specified.

**Comment by sQVe** [comment]
That doesn't work either. 😞 I even tried `keyword_pattern = [[\s+]]` and I cannot get it to show. When pressing `<C-Space>` it does though. Super confusing.

**Comment by hrsh7th** [comment]
@sQVe Did you still have,this problem?

**Comment by sQVe** [comment]
@hrsh7th Yes. I'm unable to create a `keyword_pattern` that matches empty lines or lines with only whitespace.

**Comment by abenz1267** [comment]
any news here?

**Comment by marcelarie** [comment]
any news ?

**Comment by hrsh7th** [comment]
PR welcome :)

**Comment by allan-simon** [comment]
keyword_length = 0  worked for me  and `tabnine` source

**Comment by daliusd** [comment]
This one was fun for me. I found out that in addition to cmp config I have null-ls `null_ls.builtins.completion.vsnip`. There is a reason why trying to configure vsnip in nvim-cmp does not affect vsnip for null-ls. Removing null-ls builtin solved the problem for me.
